astropy/units/format/base.py

```diff
@@ -27,11 +39,11 @@ def __init_subclass__(cls, **kwargs):
         super().__init_subclass__(**kwargs)
 
     @classmethod
-    def _get_unit_name(cls, unit):
+    def _get_unit_name(cls, unit: UnitBase) -> str:
```

Comment:
`UnitBase` doesn't have a `get_format_name` method, should it ?

Reply:
Even an abstractmethod. I think unit base is missing many such methods.

Reply:
I've been using `UnitBase` as the type for units because that is usually the best type for that, but it looks like here we should use `NamedUnit` instead.